### PR TITLE
Internal documentation [14071]

### PR DESCRIPTION
**Pull request by jparisu**
Internal documentation in Readme files to explain the directory structure and subpackages

Merge after:
- #158 
- #162

### CODE REVIEW

**Comment by jparisu** [comment]
Still in progress:

- [ ] ddsrouter_docs Documentation
- [ ] ddsrouter_yaml design explanation

File: .dev/scripts/README.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+* `SUBMODULE_PROJECT_MACROS` project macros name (usually: `$SUBMODULE_PROJECT_NAME` in capital letters). This is used for example to set the dll macro: ${SUBMODULE_PROJECT_MACROS}_DllAPI
+
+---
+
+## Design decisions
+
+### License
+
+Every file must have its license at its head.
+This license must concordance with this one (change the year to the year the file has been created):
```
Comment (rsanchez15) [inline .dev/scripts/README.md:154]:
```suggestion
This license be consistent with the following one (change the year to the year the file has been created):
```

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+```bash
+colcon build --packages-select ddsrouter_docs --cmake-args -DBUILD_DOCUMENTATION_TESTS
+colcon test --packages-select ddsrouter_docs --event-handler console_direct+
+```
+
+## Library documentation
+
+This documentation is focused on user experience.
+To learn about the repository structure, the design decisions, etc. (developer information), every package is
+commented separately and code is commented using Doxygen.
```
Comment (rsanchez15) [inline docs/README.md:27]:
```suggestion
To learn about the repository structure, design decisions, development guidelines, etc., each package is documented separately and the source code is commented using Doxygen format.
```

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# eProsima DDS Router docs
+
+This package generates the DDS Router documentation.
+[Here](https://eprosima-dds-router.readthedocs.io/en/latest/) it can be seen the online documentation hosted in
+[readthedocs](https://readthedocs.org/).
+This packages is powered by [sphinx](https://www.sphinx-doc.org/en/master/).
+
+## Documentation generation
+
+In order to install this package independently, use the following command:
```
Comment (rsanchez15) [inline docs/README.md:10]: Missing information regarding the required Python packages.

File: tools/ddsrouter_tool/README.md [hunk 4]
```diff
@@ -1 +1,47 @@
 # eProsima DDS Router Yaml Module
+
+This module create and executable that runs a DDS Router from a YAML configuration file.
```
Comment (rsanchez15) [inline tools/ddsrouter_tool/README.md:3]:
```suggestion
This module create an executable that runs a DDS Router configured via *yaml* configuration file.
```

File: tools/ddsrouter_tool/README.md [hunk 5]
```diff
@@ -1 +1,47 @@
...
+
+---
+
+## Example of use
```
Comment (rsanchez15) [inline tools/ddsrouter_tool/README.md:7]:
```suggestion
## Example of usage
```

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

File: docs/rst/user_manual/user_interface.rst [hunk 6]
```diff
@@ -81,6 +81,7 @@ It shows the usage information of the application.
     Usage: Fast DDS Router
     Connect different DDS networks via DDS through LAN or WAN.
     It will build a communication bridge between the different Participants included in the provided configuration file.
+    To close the execution gracefully use SIGINT (C^) or SIGTERM (kill).
```
[the remark this answers is not in the document]
Comment (jparisu) [reply docs/rst/user_manual/user_interface.rst:84]: This is the text as it is printed from the tool

**Review by rsanchez15 — APPROVED** [review]

**Pull request merged**